tests/scripts/test_grow_nexus.py

```diff
@@ -0,0 +1,91 @@
+
+from ess.reduce.scripts.grow_nexus import grow_nexus_file
+
+
+@pytest.fixture
+def nexus_file():
+    with tempfile.TemporaryDirectory() as tmp:
+        path = os.path.join(tmp, 'test.nxs')
+        with h5py.File(path, 'a') as hf:
+            entry = hf.create_group('entry')
+            entry.attrs['NX_class'] = 'NXentry'
+
+            instrument = entry.create_group('instrument')
+            instrument.attrs['NX_class'] = 'NXinstrument'
+
+            for group, nxclass in (
+                ('detector', 'NXdetector'),
+                ('monitor', 'NXmonitor'),
+            ):
+                detector = instrument.create_group(group)
+                detector.attrs['NX_class'] = nxclass
+
+                event_data = detector.create_group('event_data')
+                event_data.attrs['NX_class'] = 'NXevent_data'
+
+                event_data.create_dataset(
+                    'event_index',
+                    data=np.array([2, 4, 6]),
+                    maxshape=(None,),
+                    chunks=True,
+                )
+                event_data.create_dataset(
+                    'event_time_zero',
+                    data=np.array([0, 1, 2]),
+                    maxshape=(None,),
+                    chunks=True,
+                )
+                event_data.create_dataset(
+                    'event_id',
+                    data=np.array([0, 1, 2, 0, 1, 2]),
+                    maxshape=(None,),
+                    chunks=True,
+                )
+                event_data.create_dataset(
+                    'event_time_offset',
+                    data=np.array([1, 2, 1, 2, 1, 2]),
+                    maxshape=(None,),
+                    chunks=True,
+                )
+
+            yield hf
+
+
+@pytest.mark.parametrize('detector', ('detector', 'monitor'))
+def test_grow_nexus_unchanged(nexus_file, detector):
+    grow_nexus_file(nexus_file, 1, 1)
+    np.testing.assert_almost_equal(
+        [2, 4, 6],
+        nexus_file[f'entry/instrument/{detector}/event_data/event_index'],
+    )
+    np.testing.assert_almost_equal(
+        [0, 1, 2, 0, 1, 2],
+        nexus_file[f'entry/instrument/{detector}/event_data/event_id'][()],
+    )
+    np.testing.assert_almost_equal(
+        [1, 2, 1, 2, 1, 2],
+        nexus_file[f'entry/instrument/{detector}/event_data/event_time_offset'][()],
+    )
+
+
+@pytest.mark.parametrize('detector', ('detector', 'monitor'))
+def test_grow_nexus_double(nexus_file, detector):
+    grow_nexus_file(nexus_file, 2, 2)
+    np.testing.assert_almost_equal(
+        [4, 8, 12],
+        nexus_file[f'entry/instrument/{detector}/event_data/event_index'],
+    )
+    np.testing.assert_almost_equal(
+        2 * [0, 1, 2, 0, 1, 2],
+        nexus_file[f'entry/instrument/{detector}/event_data/event_id'][()],
+    )
+    np.testing.assert_almost_equal(
+        2 * [1, 2, 1, 2, 1, 2],
+        nexus_file[f'entry/instrument/{detector}/event_data/event_time_offset'][()],
+    )
```

Comment:
Suggestions:
- add a test where scaling for monitor and detector are different?
- add a test where scaling for monitor is not given?